### PR TITLE
Fix issue with system time set back

### PR DESCRIPTION
**Pull request by joaofl**
As discussed in https://github.com/systemd/systemd/issues/6036, there is an issue with systemd timers in which the next trigger will get stuck with a future date (greater than the period), whenever the system time is set-back after the timer initialization.

In this PR I amended the testsuite 53 by introducing the failing scenario, and the potential fix for the problem.

### CODE REVIEW

**Comment by joaofl** [comment]
@mrc0mmand 
Seems that all checks have passed this time. Do you know who could give a review to this PR?

File: src/core/timer.c [hunk 1]
```diff
@@ -394,6 +394,15 @@ static void timer_enter_waiting(Timer *t, bool time_change) {
                 if (v->base == TIMER_CALENDAR) {
                         usec_t b, rebased;
 
+                        /* Update last_trigger to 'now' in case
+                         * the system time changes, so that
+                         * next_elapse is not stuck with a
+                         * future date. */
+                        if (time_change){
+                                t->last_trigger.realtime = ts.realtime;
```
Comment (yuwata) [inline src/core/timer.c]: This sounds hackish for me. This intends to set `b = ts.realtime` in the below, right? How about to update the below condition like the following?
```c
if (time_change)
        b = ts.realtime;
else if (t->last_trigger.realtime > 0)
        b = t->last_trigger.realtime;
else if (state_translation_table[t->state] == UNIT_ACTIVE)
        b = UNIT(t)->inactive_exit_timestamp.realtime;
else
        b = ts.realtime;
```

File: src/core/timer.c [hunk 2]
```diff
@@ -394,6 +394,15 @@ static void timer_enter_waiting(Timer *t, bool time_change) {
                 if (v->base == TIMER_CALENDAR) {
                         usec_t b, rebased;
 
+                        /* Update last_trigger to 'now' in case
+                         * the system time changes, so that
+                         * next_elapse is not stuck with a
+                         * future date. */
```
Comment (yuwata) [inline src/core/timer.c]: Nowadays, we allow ~109ch per line. Please do not wrap lines so frequently.

**Review by yuwata — CHANGES_REQUESTED** [review]
Sorry for late review.

File: src/core/timer.c (hunk 1 above)
Comment (joaofl) [reply src/core/timer.c]: Your suggestion seems correct to me, and indeed less hacky. It also pass the tests. Thus I am resolving this comment.

File: src/core/timer.c (hunk 2 above)
Comment (joaofl) [reply src/core/timer.c]: Changed

**Comment by joaofl** [comment]
@yuwata I believe the patch is in good shape now for another check. Thanks

**Review by yuwata — APPROVED** [review]

**Comment by joaofl** [comment]
@yuwata thanks for the review.
Any more actions required from my side?

**Pull request merged**